azurerm/internal/services/costmanagement/resource_arm_cost_management_export_resource_group.go

```diff
@@ -0,0 +1,381 @@
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validateCostManagementExportContainerName,
+						},
+						"root_folder_path": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+
+			"query": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"time_frame": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(costmanagement.Custom),
+								string(costmanagement.MonthToDate),
+								string(costmanagement.TheLastMonth),
+								string(costmanagement.TheLastWeek),
+								string(costmanagement.TheLastYear),
+								string(costmanagement.WeekToDate),
+								string(costmanagement.MonthToDate),
+							}, false),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmCostManagementExportResourceGroupCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).CostManagement.ExportClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_id").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Cost Management Export Resource Group %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_cost_management_export_resource_group", *existing.ID)
+		}
+	}
+
+	from, _ := time.Parse(time.RFC3339, d.Get("recurrence_period_start").(string))
+	to, _ := time.Parse(time.RFC3339, d.Get("recurrence_period_end").(string))
+
+	status := costmanagement.Active
+	if v := d.Get("active"); !v.(bool) {
+		status = costmanagement.Inactive
+	}
+	schedule := &costmanagement.ExportSchedule{
+		Recurrence: costmanagement.RecurrenceType(d.Get("recurrence_type").(string)),
+		RecurrencePeriod: &costmanagement.ExportRecurrencePeriod{
+			From: &date.Time{Time: from},
+			To:   &date.Time{Time: to},
+		},
+		Status: status,
+	}
+
+	properties := &costmanagement.ExportProperties{
+		Schedule:     schedule,
+		DeliveryInfo: expandExportDeliveryInfo(d.Get("delivery_info").([]interface{})),
+		Format:       costmanagement.Csv,
+		Definition:   expandExportQuery(d.Get("query").([]interface{})),
+	}
+
+	account := costmanagement.Export{
+		ExportProperties: properties,
+	}
```

Comment:
Should we consolidate this to be consistent with other resources?
```suggestion
	schedule := 

	account := costmanagement.Export{
		ExportProperties: &costmanagement.ExportProperties{
			Schedule:     &costmanagement.ExportSchedule{
				Recurrence: costmanagement.RecurrenceType(d.Get("recurrence_type").(string)),
				RecurrencePeriod: &costmanagement.ExportRecurrencePeriod{
					From: &date.Time{Time: from},
					To:   &date.Time{Time: to},
				},
				Status: status,
			},
			DeliveryInfo: expandExportDeliveryInfo(d.Get("delivery_info").([]interface{})),
			Format:       costmanagement.Csv,
			Definition:   expandExportQuery(d.Get("query").([]interface{})),
		},
	}
```